models/classes/Security/DataAccess/Repository/CachedPlatformKeyChainRepository.php

```diff
@@ -51,6 +49,10 @@ public function save(KeyChain $keyChain): void
 
     public function findAll(KeyChainQuery $query): KeyChainCollection
     {
+        if ($query->getIdentifier() == null) {
```

Comment:
do you think this is the best place to handle the regeneration of the query

Reply:
the cache was not working because in most of the case the keyChainQuery didnt have identifier, that is part of cache key.
It is more a security fallback, but ofc, we should not have empty identifier

Reply:
I know we did this locally and I suggested this, but I agree with @mike-ionut-mihai-sandu-tao. In case we will have multiple keys in the future we will have an issue here. Maybe not now, so l will not block PR cause of that.